mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java

```diff
@@ -832,64 +832,154 @@ public void validatelinkedSavingsAccount(final SavingsAccount savingsAccount, fi
-                    if (expectedDisbursementDates.contains(expectedDisbursementDate)) {
-                        baseDataValidator.reset().parameter(LoanApiConstants.disbursementDateParameterName)
-                                .failWithCode(LoanApiConstants.DISBURSEMENT_DATE_UNIQUE_ERROR);
-                    }
-                    expectedDisbursementDates.add(expectedDisbursementDate);
-                    baseDataValidator.reset().parameter(LoanApiConstants.disbursementDataParameterName)
-                            .parameterAtIndexArray(LoanApiConstants.disbursementPrincipalParameterName, i).value(principal).notBlank();
-                    baseDataValidator.reset().parameter(LoanApiConstants.disbursementDataParameterName)
-                            .parameterAtIndexArray(LoanApiConstants.disbursementDateParameterName, i).value(expectedDisbursementDate)
-                            .notNull();
-                    if (principal != null) {
-                        tatalDisbursement = tatalDisbursement.add(principal);
-                    }
-                    if (expectedDisbursement.equals(expectedDisbursementDate)) {
-                        isFirstinstallmentOnExpectedDisbursementDate = true;
-                    }
-                    i++;
-                } while (i < variationArray.size());
-                if (!isFirstinstallmentOnExpectedDisbursementDate) {
-                    baseDataValidator.reset().parameter(LoanApiConstants.disbursementDateParameterName)
-                            .failWithCode(LoanApiConstants.DISBURSEMENT_DATE_START_WITH_ERROR);
-                }
-                if (tatalDisbursement.compareTo(totalPrincipal) != 0) {
-                    baseDataValidator.reset().parameter(LoanApiConstants.disbursementPrincipalParameterName)
-                            .failWithCode(LoanApiConstants.PRINCIPAL_AMOUNT_SHOULD_BE_SAME);
-                }
-                final String interestTypeParameterName = "interestType";
-                final Integer interestType = this.fromApiJsonHelper.extractIntegerSansLocaleNamed(interestTypeParameterName, element);
-                baseDataValidator.reset().parameter(interestTypeParameterName).value(interestType).ignoreIfNull()
-                        .integerSameAsNumber(InterestMethod.DECLINING_BALANCE.getValue());
-            }
-
-        }
-
-    }
+	private void validateDisbursementsAreDatewiseOrdered(JsonElement element, final DataValidatorBuilder baseDataValidator) {
+		final JsonObject topLevelJsonElement = element.getAsJsonObject();
+		final Locale locale = this.fromApiJsonHelper
+				.extractLocaleParameter(topLevelJsonElement);
+		final String dateFormat = this.fromApiJsonHelper
+				.extractDateFormatParameter(topLevelJsonElement);
+		final JsonArray variationArray = this.fromApiJsonHelper
+				.extractJsonArrayNamed(
+						LoanApiConstants.disbursementDataParameterName,
+						element);
+		for(int i=0 ; i< variationArray.size();i++){
```

Comment:
needs a null check for variationArray (else regular loans will throw a null pointer exception)